src/html_generator.py

```diff
@@ -0,0 +1,93 @@
+import tmt
+from tmt import Test, Logger, Plan
+
+
+def generate_test_html_page(test: Test, logger: Logger) -> str:
+    """
+    This function generates an HTML file with the input data for a test
+    :param test: Test object
+    :param logger: tmt.Logger instance
+    :return:
+    """
+    logger.print("Generating the HTML file...")
+    full_url = test.web_link()
+    # Adding the input data to the HTML file
+    file_html = ('''<html>
+    <head>
+    <title>HTML File</title>
+    <meta charset="UTF-8">
+    </head> 
+    <body>
+    name: ''' + str(test.name) + '''<br>
+    summary: ''' + str(test.summary) + '''<br>
+    url: <a href=\"''' + full_url + "\" target=\"_blank\">" + full_url + '''</a><br>
+    ref: ''' + str(test.fmf_id.ref) + '''<br>
+    contact: ''' + str(test.contact) + '''<br>
+    </body>
+    </html>''')
+    logger.print("HTML file generated successfully!", color="green")
+    return file_html
+
+
+def generate_plan_html_page(plan: Plan, logger: Logger) -> str:
+    """
+    This function generates an HTML file with the input data for a plan
+    :param plan: Plan object
+    :param logger: tmt.Logger instance
+    :return:
+    """
+    logger.print("Generating the HTML file...")
+    full_url = plan.web_link()
+    # Adding the input data to the HTML file
+    file_html = ('''<html>
+    <head>
+    <title>HTML File</title>
+    <meta charset="UTF-8">
+    </head> 
+    <body>
+    name: ''' + str(plan.name) + '''<br>
+    summary: ''' + str(plan.summary) + '''<br>
+    url: <a href=\"''' + full_url + "\" target=\"_blank\">" + full_url + '''</a><br>
+    ref: ''' + str(plan.fmf_id.ref) + '''<br>
+    </body>
+    </html>''')
+    logger.print("HTML file generated successfully!", color="green")
+    return file_html
+
+
+def generate_testplan_html_page(test: tmt.Test, plan: tmt.Plan, logger: Logger) -> str:
+    """
+    This function generates an HTML file with the input data for a test and a plan
+    :param test: Test object
+    :param plan: Plan object
+    :param logger: tmt.Logger instance
+    :return:
+    """
+    logger.print("Generating the HTML file...")
+    full_url_test = test.web_link()
+    full_url_plan = plan.web_link()
+    # Adding the input data to the HTML file
+    file_html = ('''<html>
```

Comment:
This is very ugly, we should talk about how to make this better from the start, not constructing HTML output this way.

Reply:
There is a separate task planned for implementing a better way to generate the static HTML page with metadata. There should also be a discussion about the best way to do this. We are currently also thinking about using Patternfly for the styling.

Reply:
Same as above, just to demonstrate the expected functionality.